### DISCUSSION TITLE
Can I use the model pb predict something without running a third package like LAMMPS?

### DISCUSSION THREAD

**Discussion by IricsDo** [deepmodeling/deepmd-kit#4043] [Q&A] [answered]
I want to use the output model of DeePMD-kit to predict something. For example, input is only coordination and output will be energy.
Like many *.pb models, it usually has an input (image, numpy, tensor) and the output will be classification, number, yes/no,...
But in graph.pb from DeePMD-kit, I don't see anywhere to give the input and get the output.

Thank for your reading .

**Comment by njzjz** [comment, marked as answer]
DeePMD-kit uses [GraphDef](https://www.tensorflow.org/api_docs/python/tf/compat/v1/GraphDef) as the protocol which contains repeated [nodes](https://www.tensorflow.org/api_docs/python/tf/compat/v1/NodeDef). All nodes can be assigned as the output. The [placeholders](https://www.tensorflow.org/api_docs/python/tf/compat/v1/placeholder) in the path of evaluating outputs are required to input.

**Comment by IricsDo** [comment]
Thanks for your quick reply. I'll try this suggestion.
